### PR TITLE
Refactor to use pagerfanta

### PR DESCRIPTION
**Pull request by dbu**
And another take, replacing #15 and #16 - now also dropping our custom pager in favor of pagerfanta.

### CODE REVIEW

**Comment by dbu** [comment]
@lsmith77 can you have a look at this now? it seems to be more sane. maybe just browse the branch in the repo, there is almost nothing left unchanged...

**Comment by lsmith77** [comment]
pushed a fix for dealing with no client configured. however I always find it unfortunate if a Bundle requires config to not explode out of the box, because it means you cannot use ``app/console config:dump-reference``

**Comment by lsmith77** [comment]
pushed another commit that removed the requirement for a client to be configured

**Comment by dbu** [comment]
> I always find it unfortunate if a Bundle requires config to not explode out of the box

agreed, good point.

> removed the requirement for a client to be configured

what does happen if i configure no client now? server error when trying to go to the search page? or already when trying to generate the search url? just having a service xy not found message is way less helpful than the configuration error imho. any good idea what we can do?

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -12,47 +12,64 @@
...
+                                ->scalarNode('search_api_url')->defaultValue('https://www.googleapis.com/customsearch/v1')->end()
+                                ->scalarNode('restrict_to_site')->defaultFalse()->end()
+                            ->end()
+                        ->end()
+                        ->arrayNode('google_cse')
+                            ->addDefaultsIfNotSet()
+                            ->canBeUnset()
+                            ->canBeEnabled()
+                            ->children()
+                                ->scalarNode('cse_id')->isRequired()->end()
```
Comment (lsmith77) [inline DependencyInjection/Configuration.php]: this does not seem to be covered in the docs

Comment (dbu) [inline DependencyInjection/Configuration.php]: oh, true that. also did not test if it still works. will do.

**Comment by dbu** [comment]
okay, i tested the frontend search now and fixed and documented it. i think we are ready to merge. care to have a last look, @lsmith77 / @thePanz ?

**Comment by thePanz** [comment]
You've done a great job! unfortunately I don't have time now to properly check the new branch (hardly planning my move to Zch? :) )
I've posted a PR with the .it.xml translation :)

**Comment by dbu** [comment]
@thePanz oh, thanks for the PR. and see you in zurich soonish then? ;-)

**Comment by lsmith77** [comment]
I also pushed a 1.0 branch before I merged this based on master

**Comment by dbu** [comment]
thanks a lot lukas!
should we tag an alpha1 already? the bundle seems to run fine in the project we are using it.

**Comment by lsmith77** [comment]
sure .. feel free!

BTW does it also still work with CmfSearchBundle?

**Comment by dbu** [comment]
i doubt it. added a note at https://github.com/symfony-cmf/SearchBundle/issues/12

**Pull request merged**